### PR TITLE
Charged atom types and sulfur groups

### PR DESCRIPTION
**Pull request by nyee**
This must be pulled in simultaneously with the RMG-Py PR also named chargedAT. **Be sure to remove the first commit redirecting the travis build before merging**

This PR adds sulfur groups for both thermo and kinetics families throughout the database. It also refactors to the new atom types defined in the RMG-Py PR.

### CODE REVIEW

**Comment by nyee** [comment]
@kehang, Could you please help me run the thermo comparison using your new script? Sorry about the large size. I would have preferred this as two separate PR's, but the sulfur was added before the atomtype refactoring, so it would be a lot of work to separate it.

File: .travis.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ before_install:
   # Add RMG-Py to PYTHONPATH
   - export PYTHONPATH=$TRAVIS_BUILD_DIR/RMG-Py:$PYTHONPATH
   - cd RMG-Py
+  - git checkout chargedAT
```
Comment (KEHANG) [inline .travis.yml]: `git checkout -b chargedAT origin/chargedAT`

**Comment by KEHANG** [comment]
The thermo validation results are promising on hydrocarbon, oxygenates, highly unsaturated hydrocarbons, and one sulfur dataset.
 
thermo estimation error (kcal/mol) | master-rmgpy & master-db | chargedAT-rmgpy & chargedAT-db 
----- | ----- | -----
small_cyclic_table | 3.42 | 3.40 
large_linear_polycyclic_table | 9.09 | 9.07
large_fused_polycyclic_table | 9.53 | 9.54
small_O_only_polycyclic_table | 5.30 | 5.30
large_linear_O_only_polycyclic_table | 6.94 | 6.94
large_fused_O_only_polycyclic_table | 10.32 | 10.31
zb_bicyclics_195_table | 9.59 | 9.59
rg_sulfur_195_table | N/A | 1.39

**Comment by alongd** [comment]
@KEHANG , this PR is now clean and ready for review. Thanks!

**Comment by alongd** [comment]
Reactions by
`P. Raghunath, N.T. Nghia, M.C. Lin, Advances in Quantum Chemistry, 2014, 69, 253-301, doi: 10.1016/B978-0-12-800345-9.00007-6 `
were commented out from the SOxNOx library (as well as from the LowT sub-library and deleted from respective training reactions) since some rates from this source seem unreasonable (one reaction has a rate O(20) lower than a calculation by Klippenstein). This resulted in a significant deviation in the NC test, which now seems to be solved. The commented out reactions are documented in #133 . I don't think this is a units error, since many are first order (s^-1), including the one with the large discrepancy I mentioned.

File: input/kinetics/families/Diels_alder_addition/groups.py [hunk 2]
```diff
@@ -24,10 +24,10 @@
     index = 1,
     label = "diene_out",
     group = """
-1 *3 Cd u0 {2,D}
-2 *4 Cd u0 {1,D} {3,S}
-3 *5 Cd u0 {2,S} {4,D}
-4 *6 Cd u0 {3,D}
+1 *3 [Cd,Od,S2d,S4d,S6d,N3d,N5d] u0 {2,D}
```
[the remark this answers is not in the document]
Comment (alongd) [reply input/kinetics/families/Diels_alder_addition/groups.py]: I saw examples where *1 is O, but not *3.
We could shrink the `diene_out` definition, and if someone finds this path important they can later add it. Would you agree?

**Comment by alongd** [comment]
I addressed the comments in fixup commits.
Let me know if it looks OK and I'll squash.

**Comment by mjohnson541** [comment]
Ok, this looks good!

**Comment by alongd** [comment]
OK, thanks!
Just squashed them in.
Once the twin -Py branch is finalized, I'll remove the tmp .travis commit here.

**Pull request merged**